snap/snapcraft.yaml

```diff
@@ -156,13 +156,30 @@ parts:
       - libuv1
     build-packages:
       - libuv1-dev
+      - pkg-config
     organize:
       usr/lib/: lib/
       include/: usr/include/
     prime:
       - lib/libdqlite*so*
       - lib/*/libuv*
       - usr/include/
+
+  k8s-binaries:
+    after: [dqlite]
+    build-snaps: [go]
+    plugin: dump
+    build-attributes: [no-patchelf]
+    source: build-scripts/
+    build-packages:
+      - git
+      - build-essential
+    override-build: |
+      set -eux
+      rm -rf go/
+      snapcraftctl build
+      bash wrapper.sh
```

Comment:
Had to put this into a script explicitly run by bash instead of sh. Not sure how this section of script worked when it was sitting in the `microk8s` part, but sh can't do `. foo.sh` or `source foo.sh`, which this script does.

Reply:
`. foo.sh` is [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#dot) so does work in `sh`.  The path locations can get confusing during the snapcraft build process, especially when splitting things into separate parts, so if I had to guess, I think you might have run up against that.

Reply:
I wouldn't mind placing the contents of the wrapper.sh here. We already run a couple of shell commands eitherway.

Reply:
@johnsca: I am able to reproduce `sh` not understanding `.` in the shell:

```
# Bash works fine
snapcraft-microk8s # cd /root/parts/etcd/build
snapcraft-microk8s ../parts/etcd/build# ls
build-certs.sh	build-k8s-binaries.sh  fetch-k8s-binaries.sh  fetch-other-binaries.sh  patches	pre-patches  set-env-binaries-location.sh  set-env-variables.sh
snapcraft-microk8s ../parts/etcd/build# . set-env-variables.sh
 ... snip working output ...

# sh doesn't
snapcraft-microk8s ../parts/etcd/build# sh
\h ../parts/etcd/build# pwd
/root/parts/etcd/build
\h ../parts/etcd/build# ls
build-certs.sh	build-k8s-binaries.sh  fetch-k8s-binaries.sh  fetch-other-binaries.sh  patches	pre-patches  set-env-binaries-location.sh  set-env-variables.sh
\h ../parts/etcd/build# . set-env-variables.sh
sh: 3: .: set-env-variables.sh: not found
```

Reply:
It turns out this is due to `dash` being the default for `sh`, and `dash` doesn't like `. foo.sh`. It instead wants `. ./foo.sh`. Still not really sure why it worked previously, other than guessing that some other package in another step switches `sh` to point at `/bin/bash`.